client/src/components/Dashboard.js

```diff
@@ -13,10 +13,51 @@ export class Dashboard extends Component {
   }
 
   componentDidMount() {
+    this.setDocuments();
+  }
+
+  setDocuments() {
     getDocuments()
       .then(documents => this.setState({ documents }))
       .catch(err => console.log(err));
   }
+  handleErrors(response) {
+    return response.json().then(json => {
+      if (!response.ok) {
+        throw json.message;
+      } else {
+        return json;
+      }
+    });
+  }
+
+  pickDocument(id) {
+    const userRequest = {
+      method: "POST",
+      body: JSON.stringify({
+        document_id: id
+      }),
+      headers: {
+        Authorization: `Bearer ${sessionStorage.getItem("token")}`,
+        "Content-Type": "application/json"
+      }
+    };
+
+    fetch("http://localhost:4000/api/translations", userRequest)
```

Comment:
Moreover, you could extract all this function in client/src/api/translations.js file

Reply:
I know it is convenient to do that but when I am trying to do so, the pickDocument  contains 2 functions which are needed in the original dashboard component (handleErrors and setDocument) which cause many problems when click on Pick Document button.